modules/core/src/v2/coins/eth.ts

```diff
@@ -1483,7 +1513,106 @@ export class Eth extends BaseCoin {
     return true;
   }
 
-  verifyTransaction(params: VerifyTransactionOptions, callback?: NodeCallback<boolean>): Bluebird<boolean> {
-    return Bluebird.resolve(true).asCallback(callback);
+  verifyCoin(txPrebuild: TransactionPrebuild): boolean {
+    return txPrebuild.coin === this.getChain();
+  }
+
+  /**
+   * Verify that a transaction prebuild complies with the original intention
+   *
+   * @param params
+   * @param params.txParams params object passed to send
+   * @param params.txPrebuild prebuild object returned by server
+   * @param params.wallet Wallet object to obtain keys to verify against
+   * @param callback
+   * @returns {boolean}
+   */
+  verifyTransaction(params: VerifyEthTransactionOptions, callback?: NodeCallback<boolean>): Bluebird<boolean> {
+    const self = this;
+    return co<boolean>(function* (): any {
+      const { txParams, txPrebuild, wallet } = params;
+      if (!txParams?.recipients || !txPrebuild?.recipients || !wallet) {
+        throw new Error(`missing params`);
+      }
+      if (txPrebuild.hopTransaction && txPrebuild.isBatch) {
+        throw new Error(`tx cannot be both a batch and hop transaction`);
+      }
+      if (txPrebuild.recipients.length !== 1) {
+        throw new Error(`txPrebuild should only have 1 recipient but ${txPrebuild.recipients.length} found`);
+      }
+      if (txPrebuild.hopTransaction) {
```

Comment:
Could these individual validation cases be broken out into separate `protected` functions (eg, `verifyHopTransaction()`, `verifyBatchTransaction()`, etc)? That should help improve readability.

Reply:
I considered splitting out the validation cases into separate functions but it seems that doing so would require me to add a bunch of extra null/undefined checks (by pushing the check that the required parameters are provided down into the validation function for each case) or else the linter would complain.

Alternatively, I would need to create new interfaces that mark parameters as required rather than optional (e.g. create a new txPrebuild interface for hop transaction validation that has the hopTransaction parameter as required instead of optional in the txPrebuild).

Either way, it seems like it trying to split out functions for each case will create a lot of extra lines of code.

Unless you have an idea for splitting out functions without having to add extra interfaces or null/undefined checks, this doesn't seem worth it.